pkg/credentials/credentials.go

```diff
@@ -221,6 +221,9 @@ func AppendCloudProviderVolumes(dpa *oadpv1alpha1.DataProtectionApplication, ds
 		}
 	}
 	for _, bslSpec := range dpa.Spec.BackupLocations {
+		if bslSpec.Velero == nil {
```

Comment:
I am unsure why are we adding this here ?  BSL can be outside the Velero context. BSL could be defined via OADP Cloud storage API as well. Even if you want to add a validation on similar lines, we should do it in validator.go. Lets remove this IMO.

Reply:
I added because of this https://github.com/openshift/oadp-operator/pull/1477#discussion_r1702337618

Per scrum, the test should not panic, but fail, as the use case is not supported

Reply:
addressed :smile: